packages/webapp/src/components/PopupFilter/PureCollapsibleSearch/index.jsx

```diff
@@ -0,0 +1,136 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { useState, useRef, useLayoutEffect } from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import styles from './styles.module.scss';
+import { ReactComponent as SearchIcon } from '../../../assets/images/search.svg';
+import Input from '../../Form/Input';
+import TextButton from '../../Form/Button/TextButton';
+import { Modal } from '../../Modals';
+
+export default function PureCollapsibleSearch({
+  value,
+  isSearchActive,
+  onChange,
+  placeholderText,
+  className,
+  containerRef,
+}) {
+  const searchRef = useRef(null);
+  const [modalStyle, setModalStyle] = useState({});
+
+  const [searchOverlayOpen, setSearchOverlayOpen] = useState(false);
+  const onSearchOpen = () => setSearchOverlayOpen(true);
+  const onSearchClose = () => setSearchOverlayOpen(false);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    onSearchClose();
+  };
+
+  // Position modal searchbar correctly
+  useLayoutEffect(() => {
+    let resizeObserver;
+
+    const updateModalStyle = (ref) => {
+      if (ref) {
+        const rect = ref.getBoundingClientRect();
+
+        setModalStyle({
+          position: 'absolute',
+          top: `${rect.top}px`,
+          left: ref === searchRef.current ? undefined : `${rect.left}px`,
```

Comment:
I don't understand the bit where it checks if the ref is actually the searchRef and sets different values in that case, can you explain that case a bit more? 🙏

Reply:
If a `containerRef` isn't provided from the parent component, this sets the position of the modal searchbar directly on top of the search button -- so it grabs a vertical position, but does not grab a horizontal position or width (as the horizontal position and width of the button itself is irrelevant). Not supplying a `left` keeps the modal default of centering horizontally, and the searchbar width will always be `95vw`.

I think overlaying the button is a pretty good default (so that a container ref doesn't always have to be created in the parent), but if it makes the code too unreadable, I can remove it. Alternatively... if you have a suggestion for writing the same in a more readable style, that would be great!

Reply:
That makes sense to me! I'd keep it and if possible just add a small comment indicating if a container ref isn't passed in the position will be calculated to overlay the button 🙂